Build failure when NEML is not present due to missing include file

## Bug Description
I get errors like this when I build BlackBear without NEML checked out:
```
blackbear/build/header_symlinks/NEMLStateAux.h:14:10: fatal error: 'neml_interface.h' file not found
#include "neml_interface.h"
         ^~~~~~~~~~~~~~~~~~
```

## Steps to Reproduce
```
git submodule deinit -f contrib/neml
make
```

## Impact
This prevents building BlackBear without NEML

--- a/include/auxkernels/NEMLStateAux.h
+++ b/include/auxkernels/NEMLStateAux.h
@@ -1,11 +1,16 @@
-//* This file is part of the MOOSE framework
-//* https://www.mooseframework.org
-//*
-//* All rights reserved, see COPYRIGHT for full restrictions
-//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
-//*
-//* Licensed under LGPL 2.1, please see LICENSE for details
-//* https://www.gnu.org/licenses/lgpl-2.1.html
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
 
 #pragma once
 

--- a/src/auxkernels/NEMLStateAux.C
+++ b/src/auxkernels/NEMLStateAux.C
@@ -1,3 +1,19 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifdef NEML_ENABLED
+
 #include "NEMLStateAux.h"
 
 registerMooseObject("BlackBearApp", NEMLStateAux);
@@ -63,3 +79,5 @@ NEMLStateAux::computeValue()
   // Trivial as we've done all the work in the constructor
   return _neml_history[_qp][_offset];
 }
+
+#endif // NEML_ENABLED

--- a/src/base/BlackBearApp.C
+++ b/src/base/BlackBearApp.C
@@ -47,7 +47,7 @@ BlackBearApp::runInputFile()
 }
 
 static void
-associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
+associateSyntaxInner(Syntax & /*syntax*/, ActionFactory & /*action_factory*/)
 {
 }
 

--- a/src/userobjects/NEMLMaterialPropertyReset.C
+++ b/src/userobjects/NEMLMaterialPropertyReset.C
@@ -71,7 +71,7 @@ NEMLMaterialPropertyReset::finalize()
 }
 
 void
-NEMLMaterialPropertyReset::threadJoin(const UserObject & y)
+NEMLMaterialPropertyReset::threadJoin(const UserObject & /*y*/)
 {
 }
 

--- a/src/userobjects/NEMLMaterialPropertyReset.C
+++ b/src/userobjects/NEMLMaterialPropertyReset.C
@@ -7,6 +7,8 @@
 //* Licensed under LGPL 2.1, please see LICENSE for details
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
+#ifdef NEML_ENABLED
+
 #include "NEMLMaterialPropertyReset.h"
 
 registerMooseObject("BlackBearApp", NEMLMaterialPropertyReset);
@@ -72,3 +74,5 @@ void
 NEMLMaterialPropertyReset::threadJoin(const UserObject & y)
 {
 }
+
+#endif // NEML_ENABLED

